packages/app-mobile/services/e2ee/crypto.ts

```diff
@@ -0,0 +1,49 @@
+import { Crypto, CryptoBuffer } from '@joplin/lib/services/e2ee/types';
+import QuickCrypto from 'react-native-quick-crypto';
+import { HashAlgorithm } from 'react-native-quick-crypto/lib/typescript/keys';
+
+const crypto: Crypto = {
+
+	getCiphers: (): string[] => {
+		return QuickCrypto.getCiphers();
+	},
+
+	getHashes: (): string[] => {
+		return QuickCrypto.getHashes();
+	},
+
+	randomBytes: async (size: number): Promise<CryptoBuffer> => {
+		return new Promise((resolve, reject) => {
+			QuickCrypto.randomBytes(size, (error, result) => {
+				if (error) {
+					reject(error);
+				} else {
+					resolve(result);
+				}
+			});
+		});
+	},
+
+	pbkdf2Raw: async (password: string, salt: CryptoBuffer, iterations: number, keylen: number, digest: string): Promise<CryptoBuffer> => {
+		const digestMap: { [key: string]: HashAlgorithm } = {
+			'sha1': 'SHA-1',
+			'sha224': 'SHA-224',
+			'sha256': 'SHA-256',
+			'sha384': 'SHA-384',
+			'sha512': 'SHA-512',
+			'ripemd160': 'RIPEMD-160',
+		};
+		const digestAlgorithm: string = digestMap[digest.toLowerCase()] || digest;
```

Comment:
Also not recommended. See https://joplinapp.org/help/dev/coding_style#dont-set-the-type-when-it-can-be-inferred

Reply:
Removed.